### DISCUSSION TITLE
Works locally but not Github Actions

### DISCUSSION THREAD

**Discussion by atomkirk** [keathley/wallaby#734] [Q&A] [answered]
I have a really simple setup with one test. My wallaby config is all the default set up exactly like the README says.

My phoenix app is an absinthe graphql api, and I serve my react front-end from a controller behind the catch-all route.

Here's an example test:

```
    session
    |> visit("/test-login/#{user.id}")
    |> visit("/conversations")
    |> print_page_html()
    |> assert_text("Conversations")
```

It works great on my local Mac. It visits `/conversations`, the react app runs and renders, the "Conversation" text appears.

On Github Actions, when I print the html after `visit("/conversations")`, it looks like it just has the static html that was returned by the backend and the react is not rendered. 

Any idea what might be going on?

**Comment by mhanberg** [comment, marked as answer]
can you confirm that in CI you have part of your build steps (either a mix alias or just a step in GHA) that bundles/builds your JS assets?

often people find it working locally because you still have your assets built from running it in dev, but on CI, it starts completely fresh each time.

**Reply by atomkirk** [reply]
yep that was it! I added a step to `mix assets.deploy` and then it worked! Is there a better command than `mix assets.deploy` or is that what you'd use?

**Reply by mhanberg** [reply]
well, it depends on what assets.deploy, is, as I suspect it's an alias you already had in your mix.exs.

a project i work on has these and we run `mix assets.build` in our github actions

![image](https://github.com/elixir-wallaby/wallaby/assets/5523984/e4c2001c-7e37-41e6-81cf-3acd9e5f512f)
